### PR TITLE
bitmex - fetch_transactions return all transactions + fix fee decimal places

### PR DESCRIPTION
**Pull request by derki-source**
if `currency: all` is omitted it returns just partial results eg. not including "deposits" at all.

adding this default parameter results in returning all results.

Also fees were off by 2 decimal places, in the `fetch_my_trades`. Changed that appropriately.

Tested on live data.

### CODE REVIEW

File: js/bitmex.js [hunk 1]
```diff
@@ -970,22 +970,24 @@ module.exports = class bitmex extends Exchange {
          */
```
Comment (carlosmiei) [inline js/bitmex.js:970]: Can you please revert all the other files except this one? All of them are automatically generated from the JS version

File: js/bitmex.js [hunk 2]
```diff
@@ -970,22 +970,24 @@ module.exports = class bitmex extends Exchange {
...
         };
         //
         //     if (since !== undefined) {
         //         // date-based pagination not supported
         //     }
         //
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['currency'] = code;
```
Comment (carlosmiei) [inline js/bitmex.js:984]: If we decide to apply this filter here we should be using `currency['id']`instead but I'm getting an error every time I try to pass a currency 
image

Did you manage to add a currency and get a valid response?

Comment (derki-source) [reply js/bitmex.js:984]: You right, I will leave that part untouched, with just added `currency: all` as default for request.

File: js/bitmex.js (hunk 1 above)
Comment (derki-source) [reply js/bitmex.js:970]: good to know, reverted.

File: js/bitmex.js [hunk 3]
```diff
@@ -1538,7 +1539,7 @@ module.exports = class bitmex extends Exchange {
         const side = this.safeStringLower (trade, 'side');
         // price * amount doesn't work for all symbols (e.g. XBT, ETH)
         let fee = undefined;
-        const feeCostString = Precise.stringDiv (this.safeString (trade, 'execComm'), '1e8');
+        const feeCostString = Precise.stringDiv (this.safeString (trade, 'execComm'), '1e6');
```
[the remark this answers is not in the document]
Comment (derki-source) [reply js/bitmex.js]: most of the trading pairs are with usdt but yes if you do those few outside usdt, they follow different decimal rule. I reverted it

**Comment by carlosmiei** [comment]
✅ ✅ 
```
p bitmex fetchTransactions                               
Python v3.9.7
CCXT v1.91.60
bitmex.fetchTransactions()
[{'address': None,
  'addressFrom': None,
  'addressTo': None,
  'amount': 32.0,
  'comment': None,
  'currency': 'USDT',
  'datetime': '2022-07-27T16:52:53.745Z',
  'fee': {'cost': None, 'currency': 'USDT'},
  'id': '584b89ef-cda5-7a9c-c1f2-2955d9cf1656',
  'info': {'account': '2059056',
           'address': '',
           'amount': '32000000',
           'currency': 'USDt',
           'fee': None,
           'marginBalance': None,
           'text': '',
           'timestamp': '2022-07-27T16:52:53.745Z',
           'transactID': '584b89ef-cda5-7a9c-c1f2-2955d9cf1656',
           'transactStatus': 'Completed',
           'transactTime': '2022-07-27T16:52:53.745Z',
           'transactType': 'Deposit',
           'tx': '',
           'walletBalance': '32000000'},
  'network': None,
  'status': 'ok',
  'tag': None,
  'tagFrom': None,
  'tagTo': None,
  'timestamp': 1658940773745,
  'txid': None,
  'type': 'deposit',
  'updated': 1658940773745}]
  ```

**Review by carlosmiei — APPROVED** [review]

**Review by carlosmiei — APPROVED** [review]

File: js/bitmex.js [hunk 4]
```diff
@@ -1022,6 +1023,7 @@ module.exports = class bitmex extends Exchange {
         //   }
         //
         const id = this.safeString (transaction, 'transactID');
+        currency = this.safeCurrencyCode (this.safeString (transaction, 'currency'));
```
Comment (kroitor) [inline js/bitmex.js:1026]: This looks a bit wrong, i think it should not overwrite the `currency` structure with a currency code here.

Comment (frosty00) [reply js/bitmex.js:1026]: yeah we can use safeCurrency instead

Comment (kroitor) [inline js/bitmex.js:1026]:
```suggestion
        currency = this.safeCurrency (this.safeString (transaction, 'currency'));
```

Comment (carlosmiei) [reply js/bitmex.js]: @kroitor in this case, the currency is available in the response, we can't ignore that as well

File: js/bitmex.js [hunk 5]
```diff
@@ -1037,12 +1039,13 @@ module.exports = class bitmex extends Exchange {
...
-        amountString = Precise.stringDiv (Precise.stringAbs (amountString), '1e8');
+        const scale = (currency === 'BTC') ? '1e8' : '1e6';
+        amountString = Precise.stringDiv (Precise.stringAbs (amountString), scale);
         let feeCostString = this.safeString (transaction, 'fee');
-        feeCostString = Precise.stringDiv (feeCostString, '1e8');
+        feeCostString = Precise.stringDiv (feeCostString, scale);
         const fee = {
             'cost': this.parseNumber (feeCostString),
-            'currency': 'BTC',
+            'currency': currency,
```
Comment (kroitor) [inline js/bitmex.js:1048]:
```suggestion
            'currency': currency['code'],
```

File: js/bitmex.js [hunk 6]
```diff
@@ -1063,8 +1066,7 @@ module.exports = class bitmex extends Exchange {
             'tagTo': undefined,
             'type': type,
             'amount': this.parseNumber (amountString),
-            // BTC is the only currency on Bitmex
-            'currency': 'BTC',
+            'currency': currency,
```
Comment (kroitor) [inline js/bitmex.js:1069]:
```suggestion
            'currency': currency['code'],
```

File: js/bitmex.js (hunk 4 above)
Comment (kroitor) [reply js/bitmex.js]: I've added fixes for it.

File: js/bitmex.js [hunk 7]
```diff
@@ -1037,12 +1039,13 @@ module.exports = class bitmex extends Exchange {
             addressTo = address;
         }
         let amountString = this.safeString (transaction, 'amount');
-        amountString = Precise.stringDiv (Precise.stringAbs (amountString), '1e8');
+        const scale = (currency === 'BTC') ? '1e8' : '1e6';
```
Comment (kroitor) [inline js/bitmex.js:1042]:
```suggestion
        const scale = (currency['code'] === 'BTC') ? '1e8' : '1e6';
```

File: js/bitmex.js [hunk 8]
```diff
@@ -1022,6 +1023,7 @@ module.exports = class bitmex extends Exchange {
         //   }
         //
         const id = this.safeString (transaction, 'transactID');
+        currency = this.safeCurrency (this.safeString (transaction, 'currency'));
```
Comment (kroitor) [inline js/bitmex.js:1026]:
```suggestion
        const currencyId = this.safeString (transaction, 'currency');
        currency = this.safeCurrency (currencyId, currency);
```

File: js/bitmex.js (hunk 4 above)
Comment (carlosmiei) [reply js/bitmex.js]: @kroitor nice thanks

**Pull request merged**